### PR TITLE
adding thorchain chains

### PR DESCRIPTION
**Pull request by flipside-kellen**
Added THORChain, broken down by chains, using the Flipside Crypto API. Also added Dogecoin, Litecoin, and Bitcoin Cash to the chains.json file, as they were new to DeFi Llama.

### CODE REVIEW

**Comment by waynebruce0x** [comment]
Can this be done using on chain calls instead?

**Comment by flipside-kellen** [comment]
@waynebruce0x I can ask around. Is there an issue with using the Flipside API?

**Comment by g1nt0ki** [comment]
@flipside-kellen hi kellen, nothing against flipside, tracking bridges are tricky. Main contention is where to count the locked tokens, either on the source side or on destination, if you are in our discord, you can look at synapse teams comments here: https://discord.com/channels/823822164956151810/953968487687135233 

tl;dr: they want to count at destination. There is pros and cons to both, though I personally prefer counting tokens on the source chain for evm chains, even to me, it feels little odd to list dogecoin or litecoin as a chain and show chain tvl there since they dont even support defi

**Comment by kblumberg** [comment]
Hi @g1nt0ki, thanks for the feedback - just read through the messages in the discord. To provide a little context on how THORChain works... It is a truly native cross-chain dex, meaning that there is no minting or burning. LP-ers can provide liquidity in the native asset to the pool and the THORChain nodes track that activity. So when someone on DOGE or LTC provides liquidity, it truly is liquidity providing on the native chain, which I believe makes THORChain quite unique.

Happy to provide more details or hop in the discord to discuss.

**Comment by g1nt0ki** [comment]
@kblumberg sounds good, can you find ulysses from defillama discord and send him a dm?

**Comment by kblumberg** [comment]
@g1nt0ki awesome, will do.

**Comment by llamatester** [comment]
The adapter at projects/thorchain exports TVL: 
        
 
 
    thorchain                 236.46 M
    thorchain-staking         266.19 M
    staking                   266.19 M
    
    total                    236.32 M

**Comment by llamatester** [comment]
The adapter at projects/thorchain exports TVL: 
        
 
 
    thorchain-staking         264.65 M
    thorchain                 235.94 M
    staking                   264.65 M
    
    total                    235.94 M

**Comment by llamatester** [comment]
The adapter at projects/thorchain exports TVL: 
        
 
 
    thorchain                 235.64 M
    thorchain-staking         264.65 M
    staking                   264.65 M
    
    total                    235.64 M

**Comment by g1nt0ki** [comment]
@flipside-kellen I have made some changes to the code. Call to flipside only once and cache the response.

Also, commented out Terra because their chain is... unstable now.

Woud merge it if the changes are fine

**Comment by kblumberg** [comment]
@g1nt0ki that sounds great - thanks for making those edit and yes, those changes are good with me.
